src/v/cluster/partition.cc

```diff
@@ -112,14 +117,19 @@ partition::partition(
                 _cloud_storage_partition
                   = ss::make_shared<cloud_storage::remote_partition>(
                     _archival_meta_stm->manifest(),
-                    cloud_storage_api.local(),
+                    _cloud_storage_api.local(),
                     cloud_storage_cache.local(),
                     s3::bucket_name{*bucket});
             }
         }
+
+        // Construct cloud_storage write path (ntp_archiver)
+        maybe_construct_archiver();
     }
 }
 
+partition::~partition() {}
```

Comment:
Can we just use the default destructor?

Reply:
Defining the destructor in the .cc enables me to avoid `#include` of the ntp_archiver_service.h in `partition.h` -- just a little bit of header hygiene, using fwd.h when the class is only holding pointers rather than instances.